lib/if.c

```diff
@@ -111,15 +113,27 @@ int if_cmp_name_func(const char *p1, const char *p2)
 		if (!*p2)
 			return 1;
 
-		x1 = strtol(p1, (char **)&p1, 10);
-		x2 = strtol(p2, (char **)&p2, 10);
+		x1 = strtol(p1, (char **)&tmp1, 10);
+		x2 = strtol(p2, (char **)&tmp2, 10);
 
 		/* let's compare numbers now */
 		if (x1 < x2)
 			return -1;
 		if (x1 > x2)
 			return 1;
 
+		/* Compare string if numbers are equal (distinguish foo-1 from
+		 * foo-001)
+		 */
+		l1 = strspn(p1, "0123456789");
+		l2 = strspn(p2, "0123456789");
```

Comment:
It's enough to have `strspn(x, "0")`. We already checked that the numbers are equal, the only difference here may be the number of zeroes.

Reply:
You are probably right, but I think it's better to get the whole number, to ensure that no other corner case is missing. It is balanced with the previous call to strcspn().

Reply:
+1 with @NicolasDichtel

Reply:
Not sure which corner cases can happen here, to be honest. But the only harm of using the whole string is an unnoticeably smaller performance, so I won't insist.